test/test-fs.c

```diff
@@ -3889,6 +3891,12 @@ TEST_IMPL(fs_read_dir) {
    * created on. That is why this assertion is a bit lenient.
    */
   ASSERT((r >= 0) || (r == UV_EISDIR));
+#elif defined(__QNX__)
+  /*
+   * If UV_FS_O_DIRECTORY is supplied, QNX would return ENOSYS. Otherwise
+   * UV_EISDIR would be returned. Here we are lenient to accept both.
```

Comment:
"would return" reads like you're not sure. Present tense is better:
```suggestion
   * If UV_FS_O_DIRECTORY is supplied, QNX returns ENOSYS. Otherwise
   * UV_EISDIR is returned. Here we are lenient and accept both.
```